zrml/prediction-markets/src/lib.rs

```diff
@@ -130,26 +191,28 @@ mod pallet {
             // Slash outstanding bonds; see
             // https://github.com/zeitgeistpm/runtime-audit-1/issues/34#issuecomment-1120187097 for
             // details.
-            let slash_market_creator = |amount| {
-                T::AssetManager::slash_reserved_named(
-                    &Self::reserve_id(),
-                    Asset::Ztg,
-                    &market.creator,
-                    amount,
-                );
+            match &market.bonds.advisory {
+                Some(bond) if !bond.is_settled => {
+                    Self::slash_advisory_bond(&market_id, &market, true)?;
+                }
+                _ => (),
             };
+            match &market.bonds.oracle {
+                Some(bond) if !bond.is_settled => {
+                    Self::slash_oracle_bond(&market_id, &market)?;
+                }
+                _ => (),
+            };
+            match &market.bonds.validity {
+                Some(bond) if !bond.is_settled => {
+                    Self::slash_validity_bond(&market_id, &market)?;
+                }
+                _ => (),
+            };
```

Comment:
Nice idea. Tried it but since you can't use generics for closures, you need to put the function into a `Box`, etc. so I don't think this will improve readability. If we get more of these, we can use another macro ;)